log/src/main/java/br/com/codenation/log/repository/LogRepository.java

```diff
@@ -12,16 +10,53 @@
 
 public interface LogRepository extends JpaRepository<Log, Long> {
 
-    Optional<Log> findByAmbienteAndId(Ambiente ambiente, Long id);
-
-    List<Log> findAllByAmbiente(Ambiente ambiente, Sort sort);
-
-    List<Log> findAllByAmbienteAndNivel(Ambiente ambiente, Nivel nivel, Sort sort);
-
-    List<Log> findAllByAmbienteAndDescricao(Ambiente ambiente, String decricao, Sort sort);
-
-    @Query(value = "SELECT * FROM log WHERE ambiente=?1 AND payload ->> 'origem'= ?2 ORDER BY ?3", nativeQuery = true)
-    List<Log> findAllByAmbienteAndOrigem(Ambiente ambiente, String origem, String ordenacao);
-
-    void deleteByAmbienteAndId(Ambiente ambiente, Long id);
+    @Query(value = "SELECT " +
+            "ambiente, " +
+            "nivel, " +
+            "cast (payload AS VARCHAR), " +
+            "usuario_id AS usuarioId, " +
+            "count(*) OVER (PARTITION BY descricao, nivel) AS frequencia " +
+            "FROM log " +
+            "WHERE id=?1", nativeQuery = true)
```

Comment:
```suggestion
            "WHERE id=:id", nativeQuery = true)
```
Prefiro usar parâmetro nomeado do que posicional. Isso garante que mesmo mudando a posição dos parâmetros você ainda estará passando os dados corretos.
Pode replicar isso nos demais métodos também.